### PR TITLE
Stop using ``pkg_resources`` for Salt's entrypoints loading. Even on Py3.5

### LINKED ISSUE #60085: [BUG] pip.installed fails with  TypeError: 'NoneType' object is not reversible in pip_state.py
**Description**
State for pip.installed fails with python exception 

State is this:

salthelpers-scripts-upgrade-global-packages:
  pip.installed:
    - names:
      - awscli >= 1.11.34
      - boto3 >= 1.4.1
    - upgrade: True

OS: Ubuntu 20.04
Salt ver: 3003 , installed from official PPA.
See detailed salt versions in the end of the report.

Error is:
TypeError: 'NoneType' object is not reversible
as below:

File "/usr/lib/python3/dist-packages/salt/states/pip_state.py", line 324, in _check_if_installed
                  for version in reversed(available_versions):
              TypeError: 'NoneType' object is not reversible

**Setup**
State that fails is described above.

**Steps to Reproduce the behavior**
Create SLS with state above and apply it.
(I applied with salt-call --local state.highstate,
However, I'm pretty sure you can apply just this particular state 
and you will shall get same issue).

You will get this output for the state:

ID: salthelpers-scripts-upgrade-global-packages
    Function: pip.installed
        Name: boto3 >= 1.4.1
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python3/dist-packages/salt/state.py", line 2171, in call
                  ret = self.states[cdata["full"]](
                File "/usr/lib/python3/dist-packages/salt/loader.py", line 1235, in __call__
                  return self.loader.run(run_func, *args, **kwargs)
                File "/usr/lib/python3/dist-packages/salt/loader.py", line 2268, in run
                  return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
                File "/usr/lib/python3/dist-packages/salt/loader.py", line 2283, in _run_as
                  return _func_or_method(*args, **kwargs)
                File "/usr/lib/python3/dist-packages/salt/loader.py", line 2316, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python3/dist-packages/salt/states/pip_state.py", line 852, in installed
                  out = _check_if_installed(
                File "/usr/lib/python3/dist-packages/salt/states/pip_state.py", line 324, in _check_if_installed
                  for version in reversed(available_versions):
              TypeError: 'NoneType' object is not reversible
     Started: 17:27:26.200409
    Duration: 1212.3 ms
     Changes:

There will be one more output for aws-cli but almost identical to shown above.

I'm able to workaround the bug by adding option:
  - force_reinstall: True
 
This makes code going different route and this particular bug does not trigger.
However, I do not feel it's good idea to do the workaround everywhere.

**Expected behavior**
Expected behavior is to have state successful with Result: True

**Versions Report**
<details><summary>salt --versions-report</summary>

Salt Version:
          Salt: 3003

Dependency Versions:
          cffi: 1.14.5
      cherrypy: Not Installed
      dateutil: 2.7.3
     docker-py: Not Installed
         gitdb: 2.0.6
     gitpython: 3.0.7
        Jinja2: 2.10.1
       libgit2: Not Installed
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 0.6.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: 2.20
      pycrypto: Not Installed
  pycryptodome: 3.6.1
        pygit2: Not Installed
        Python: 3.8.5 (default, Jan 27 2021, 15:41:15)
  python-gnupg: 0.4.5
        PyYAML: 5.3.1
         PyZMQ: 18.1.1
         smmap: 2.0.5
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.2

System Versions:
          dist: ubuntu 20.04 focal
        locale: utf-8
       machine: x86_64
       release: 5.4.0-72-generic
        system: Linux
       version: Ubuntu 20.04 focal

```
PASTE HERE
```
</details>

**Additional context**
May be I'm missing something simple, because bug seems shall be discovered by many people
but I did not find anything new and relevant by googling.
If you have any ideas on what I can be missing, let me know.

### PR DESCRIPTION
**Pull request by s0undt3ch**
### What does this PR do?
See title

This should fix the [Debian 9 CI Failures](https://jenkins.saltproject.io/job/pr-debian-9-amd64-py3-pytest-slow/job/master)
Fixes #60085

### CODE REVIEW

**Comment by s0undt3ch** [comment]
re-run full all

**Review by waynew — DISMISSED** [review]

**Comment by s0undt3ch** [comment]
re-run full all

**Comment by s0undt3ch** [comment]
re-run full all

**Review by waynew — DISMISSED** [review]

**Comment by s0undt3ch** [comment]
re-run full all

**Comment by s0undt3ch** [comment]
re-run full all

**Comment by s0undt3ch** [comment]
re-run full all

**Comment by s0undt3ch** [comment]
re-run full all

File: salt/ext/importlib_metadata/__init__.py [hunk 1]
```diff
@@ -41,15 +41,22 @@
     NullFinder,
     PyPy_repr,
     install,
-    Protocol,
+    # Protocol,
```
Comment (Ch3LL) [inline salt/ext/importlib_metadata/__init__.py:44]: removed?

File: salt/ext/importlib_metadata/__init__.py [hunk 2]
```diff
@@ -41,15 +41,22 @@
...
 )
 
 from configparser import ConfigParser
 from contextlib import suppress
 from importlib import import_module
 from importlib.abc import MetaPathFinder
 from itertools import starmap
-from typing import Any, List, Optional, TypeVar, Union
+#from typing import Any, List, Optional, TypeVar, Union
```
Comment (Ch3LL) [inline salt/ext/importlib_metadata/__init__.py:52]: removed?

File: salt/ext/importlib_metadata/__init__.py [hunk 3]
```diff
@@ -198,20 +205,36 @@ def __repr__(self):
         return '<FileHash mode: {} value: {}>'.format(self.mode, self.value)
 
 
-_T = TypeVar("_T")
+# _T = TypeVar("_T")
```
Comment (Ch3LL) [inline salt/ext/importlib_metadata/__init__.py:208]: Why is this line and the ones below commented?

File: salt/ext/importlib_metadata/__init__.py [hunk 4]
```diff
@@ -300,8 +326,25 @@ def _local(cls, root='.'):
         )
         return PathDistribution(zipp.Path(meta.build_as_zip(builder)))
 
+#     @property
```
Comment (Ch3LL) [inline salt/ext/importlib_metadata/__init__.py:329]: here too?

File: salt/ext/importlib_metadata/__init__.py [hunk 5]
```diff
@@ -637,7 +680,16 @@ def distributions(**kwargs):
     return Distribution.discover(**kwargs)
 
 
-def metadata(distribution_name) -> PackageMetadata:
+# def metadata(distribution_name) -> PackageMetadata:
```
Comment (Ch3LL) [inline salt/ext/importlib_metadata/__init__.py:683]: and here

File: salt/ext/importlib_metadata/_compat.py [hunk 6]
```diff
@@ -22,18 +22,19 @@
...
-    pytest-mypy complains here because:
-    error: Incompatible import of "Protocol" (imported name has type
-    "typing_extensions._SpecialForm", local name has type "typing._SpecialForm")
-    """
-    from salt.ext.typing_extensions import Protocol  # type: ignore
+# __all__ = ['install', 'NullFinder', 'PyPy_repr', 'Protocol']
+__all__ = ['install', 'NullFinder', 'PyPy_repr']
+
+
+# try:
```
Comment (Ch3LL) [inline salt/ext/importlib_metadata/_compat.py:29]: and here

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: salt/ext/importlib_metadata/__init__.py (hunk 1 above)
Comment (s0undt3ch) [reply salt/ext/importlib_metadata/__init__.py:44]: Yes, salt currently makes no use of typing, specially on `salt/ext` and typing on 3.5 also reduces performance.

File: salt/ext/importlib_metadata/__init__.py (hunk 2 above)
Comment (s0undt3ch) [reply salt/ext/importlib_metadata/__init__.py:52]: Same as above, and yes most of the code was left as comment

**Review by s0undt3ch — COMMENTED** [review]
The code that was commented out is so that it doesn't deviate much from upstream code.
However, removing typing, specially on Py3.5 does give us some performance benefits, and we're not using typing information, specially in `salt/ext`

**Review by Ch3LL — DISMISSED** [review]

**Comment by s0undt3ch** [comment]
re-run full all

**Pull request merged**